consumer/resolver.go

```diff
@@ -279,10 +279,25 @@ func (r *Resolver) updateLocalShards() {
 	var next = make(map[pc.ShardID]*shard, len(r.state.LocalItems))
 
 	for _, li := range r.state.LocalItems {
-		var assignment = li.Assignments[li.Index]
-		var id = pc.ShardID(li.Item.Decoded.(allocator.Item).ID)
+		var (
+			asn       = li.Assignments[li.Index]
+			id        = pc.ShardID(li.Item.Decoded.(allocator.Item).ID)
+			shard, ok = r.shards[id]
+		)
+
+		if !ok {
+			// Local shard doesn't exist and must be created.
+		} else if p := shard.resolved.assignment; asn.Decoded.(allocator.Assignment).Slot < p.Decoded.(allocator.Assignment).Slot {
+			// Local shard is being promoted (update).
+		} else if p.Raw.CreateRevision == asn.Raw.CreateRevision {
+			// Assignment value was updated but the assignment itself is unchanged.
```

Comment:
I'm not sure I understand this case. How would the assignment value be updated without creating a new revision?

Reply:
[see the etcd spec](https://github.com/etcd-io/etcd/blob/main/api/mvccpb/kv.proto#L12-L29). CreateRevision is separately tracked from ModRevision.